### PR TITLE
Add reverse proxy plugin integration test

### PR DESCRIPTION
**Pull request by gtrkiller**
Applicable spec: <link>

### Overview

Add an integration test for reverse-proxy-auth-plugin plugin.

### Rationale

Keeping track of frequently used plugins' functionality.

### Juju Events Changes

N/A

### Module Changes

N/A

### Library Changes

N/A

### Checklist

- [X] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [X] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [X] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [X] The documentation is generated using `src-docs`
- [X] The documentation for charmhub is updated.
- [X] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: tests/integration/test_plugins.py [hunk 1]
```diff
@@ -389,6 +389,22 @@ async def test_docker_build_publish_plugin(unit_web_client: UnitWebClient):
     ), f"docker-build-publish configuration option not found. {config_page}"
 
 
+async def test_reverse_proxy_plugin(unit_web_client: UnitWebClient):
```
Comment (arturo-seijas) [inline tests/integration/test_plugins.py:392]: According to our contribution guide, the code blocks corresponding to the arrange, assert and act should be separated by a blank line. See https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#test-structure

**Review by arturo-seijas — APPROVED** [review]

**Review by arturo-seijas — APPROVED** [review]

File: jenkins_rock/rockcraft.yaml [hunk 2]
```diff
@@ -45,7 +45,7 @@ parts:
       - default-jre-headless
       - git
     build-environment:
-      - JENKINS_VERSION: 2.427
+      - JENKINS_VERSION: 2.426.3
```
Comment (arturo-seijas) [inline jenkins_rock/rockcraft.yaml:48]: Why's this?

Comment (gtrkiller) [reply jenkins_rock/rockcraft.yaml:48]: Trivy is complaining about 2.427 now

Comment (gtrkiller) [reply jenkins_rock/rockcraft.yaml:48]: 2.426.3 has the fix for the problem Trivy complains about, and is also the latest LTS version

Comment (arturo-seijas) [reply jenkins_rock/rockcraft.yaml:48]: Understood. Can you open a different PR for this change so that it can be easily tracked?

**Review by arturo-seijas — APPROVED** [review]

**Review by yanksyoon — APPROVED** [review]

**Pull request merged**